snapcraft/plugins/v2/python.py

```diff
@@ -0,0 +1,175 @@
+This plugin uses the common plugin keywords as well as those for "sources".
+For more information check the 'plugins' topic for the former and the
+'sources' topic for the latter.
+
+Additionally, this plugin uses the following plugin-specific keywords:
+
+    - requirements:
+      (list of strings)
+      List of paths to requirements files.
+    - constraints:
+      (list of strings)
+      List of paths to constraint files.
+    - python-packages:
+      (list)
+      A list of dependencies to get from PyPI. If needed, pip,
+      setuptools and wheel can be upgraded here.
+
+This plugin also interprets these specific build-environment entries:
+
+    - SNAPCRAFT_PYTHON_INTERPRETER
+      (default: python3)
+      The interpreter binary to search for in PATH.
+    - SNAPCRAFT_PYTHON_VENV_ARGS
+      Additional arguments for venv.
+
+By default this plugin uses python from the base, if a part using
+this plugin uses a build-base other than that of the base, or a
+different interpreter is desired, it must be bundled in the snap
+(including venv) and must be in PATH.
+
+It is required to bundle python when creating a snap that uses
+classic confinement, this can be accomplished on Ubuntu by
+adding stage-packages (i.e.; python3-venv).
+
+Use of python3-<python-package> in stage-packages will force the
+inclusion of the python interpreter.
+"""
+
+from textwrap import dedent
+from typing import Any, Dict, List, Set
+
+from snapcraft.plugins.v2 import PluginV2
+
+
+class PythonPlugin(PluginV2):
+    @classmethod
+    def get_schema(cls) -> Dict[str, Any]:
+        return {
+            "$schema": "http://json-schema.org/draft-04/schema#",
+            "type": "object",
+            "additionalProperties": False,
+            "properties": {
+                "requirements": {
+                    "type": "array",
+                    "minitems": 1,
+                    "uniqueItems": True,
+                    "items": {"type": "string"},
+                    "default": [],
+                },
+                "constraints": {
+                    "type": "array",
+                    "minitems": 1,
+                    "uniqueItems": True,
+                    "items": {"type": "string"},
+                    "default": [],
+                },
+                "python-packages": {
+                    "type": "array",
+                    "minitems": 1,
+                    "uniqueItems": True,
+                    "items": {"type": "string"},
+                    "default": [],
+                },
+            },
+        }
+
+    def get_build_packages(self) -> Set[str]:
+        return {"findutils", "python3-dev", "python3-venv"}
+
+    def get_build_environment(self) -> Dict[str, str]:
+        return {
+            "SNAPCRAFT_PYTHON_INTERPRETER": "python3",
+            "SNAPCRAFT_PYTHON_VENV_ARGS": "",
+        }
+
+    def get_build_commands(self) -> List[str]:
+        build_commands = [
+            '"${SNAPCRAFT_PYTHON_INTERPRETER}" '
```

Comment:
In general I think it's fine to have a long string on a single line where it helps readability...  unless we intend to strictly enforce length?

Reply:
ok, I don't mind changing, just mark the ones you think will be improved by single line, or is it all of them?

Reply:
fixed what I suspect is the offending line

Reply:
All good with what you have, thanks. :)